src/libexpr/primops/fetchTree.cc

```diff
@@ -68,7 +69,16 @@ void emitTreeAttrs(
 std::string fixURI(std::string uri, EvalState & state, const std::string & defaultScheme = "file")
```

Comment:
Yikes. What is this about? URIs can't be fixed.

I wish we'd just use (semi-)structured data. URIs are such a ~mess~ burden.

Reply:
This is the code that changes passed paths and strings to `fetchTree` to be an URL.

Examples: `"git@asdf:foo.git"`, `./.`, `"https://github.com/nixos/nixpkgs`

Sadly, `fetchers::Input::fromAttrs` expects to receive a standard attrset with an url-formatted string, so this goes through `ParsedURL` twice.

Reply:
> a standard attrset with an url-formatted string

That should never have been a thing. When working with any kind of serialization, including URIs, we must keep it simple by only having our data in one of two states: serialized, or deserialized. Doing string operations on URIs is asking for trouble.